src/plugins/lua/replies.lua

```diff
@@ -91,16 +101,133 @@ local function replies_check(task)
+    local last_score = -1
+
+    -- getting last score of recipients
+    local function redis_zrange_cb(err, data)
+      if err ~= nil then
+        rspamd_logger.errx(task,
+                'redis_zrange_cb error when reading zrange withscores from global replies set with error: %s', err)
+        return
+      end
+      last_score = tonumber(data[#data])
+      rspamd_logger.infox(task, 'last score %s of global replies set was received', last_score)
+
+      -- if last score wasn't found
+      if last_score == -1 or last_score == nil then
+        rspamd_logger.errx(task, 'have not found any senders in global replies set, considering last score as 0')
+        last_score = 0
+      end
+
+      -- updating params considering last score of existing sender
+      for i = 1, #params, 2 do
+        params[i] = params[i] + last_score
+      end
+
+      add_to_global_replies_set(params, sender_key)
+    end
+
+    lua_util.debugm(N, task, 'Getting recipients withscores from global replies set to get last score')
+
+    -- getting scores of recipients connected to sender
+    lua_redis.redis_make_request(task,
+            redis_params,
+            sender_key,
+            false,
+            redis_zrange_cb,
+            'ZRANGE',
+            {'rsrk_verified_recipients', '0', '-1', 'WITHSCORES'}
+    )
+  end
+
+  local function add_to_replies_set(recipients)
+    local sender = task:get_reply_sender()
+
+    local task_time = task:get_timeval(true)
+
+    local params = {}
+    -- making params out of recipients list for replies set
+    local j = 1
+    for i = 1, #recipients * 2, 2 do
+      table.insert(params, i, tostring(task_time))
+      table.insert(params, i + 1, tostring(recipients[j]))
+      j = j + 1
+    end
+
+    local sender_string = lua_util.maybe_obfuscate_string(tostring(sender), settings, settings.sender_prefix)
+    local sender_key = make_key(sender_string:lower(), 8)
+
+    lua_util.debugm(N, task, 'Adding  recipients %s to sender %s local replies set', params, sender_key)
+
+    table.insert(params, 1, sender_key)
+
+    local function zadd_cb(err, data)
+      if err ~= nil then
+        rspamd_logger.errx(task, 'adding to %s failed with error: %s', sender_key, err)
+        return
+      end
+
+      table.remove(params, 1)
+
+      rspamd_logger.infox(task, 'added data: %s to sender: %s with code: %s', params, sender_key, data)
+
+      update_global_replies_set(params, sender_key)
+
+    end
+
+    local _, conn, _ = lua_redis.redis_make_request(task, -- making local replies set (sender - recipients)
+            redis_params, -- connect params
+            sender_key, -- hash key
+            true, -- is write
+            zadd_cb, --callback
+            'ZADD', -- command
+            params -- arguments
+    )
+    -- adding expiration to the local replies set
+    conn:add_cmd('EXPIRE', { sender_key, tostring(math.floor(settings['expire'] * 1000)) })
```

Comment:
Not sure why do we multiply it by 1000. Time values in Rspamd are usually stored as plain ucl variables, such as `1h` or `1min`.